asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -16,25 +17,29 @@
 
 public class Processor {
 
-	protected RubyHash config;
-    protected Ruby rubyRuntime;
+	protected Map<String, Object> config;
 
     public Processor(Map<String, Object> config) {
-        this.rubyRuntime = JRubyRuntimeContext.get();
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
+        this.config = config;
     }
 
-    public void update_config(Map<Object, Object> config) {
+    public void update_config(Map<String, Object> config) {
     	this.config.putAll(config);
     }
     
-    public Map<Object, Object> getConfig() {
+    public Map<String, Object> getConfig() {
     	return this.config;
     }
 
-    public RubyHash setConfig(Map<String, Object> config) {
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
-        return null;
+    public void setConfig(Map<String, Object> config) {
+        if (config instanceof RubyHashMapDecorator) {
+            // ProcessorProxy has already taken over, replace config
+            this.config.clear();
+            this.config.putAll(config);
+        } else {
+            // Processor is still independent, simply set the member
+            this.config = config;
+        }
```

Comment:
It will be hard to make this method independent of Ruby if we really decide to factor out an API module.
Any ideas are welcome.